torch/distributed/pipelining/schedules.py

```diff
@@ -944,6 +937,12 @@ def step(self, *args, target=None, losses: Optional[List] = None, **kwargs):
         # Split inputs into microbatches
         args_split, kwargs_split = self._split_inputs(args, kwargs)
 
+        # Check if buffers initialized for communication
+        for stage in self._stages:
+            if not stage.buffers_initialized:
+                logger.debug("init_buffers for %s", stage.stage_index)
+                stage.init_buffers(self._n_microbatches, args_split[0], kwargs_split[0])
```

Comment:
is it safe to call init_buffers one stage at a time for looped schedules?  i guess, this would work for 'looped' schedules but fail for V-shaped schedules?  bc stage.init_buffers will block until stage+1.init_buffers is called, right?